### PR TITLE
Can disable the modification of SSH settings in BTCPay Server

### PR DESCRIPTION
**Pull request by NicolasDorier**
Implement https://github.com/btcpayserver/btcpayserver/issues/1810

image

image

image

### CODE REVIEW

**Review by Kukks — APPROVED** [review]
ack

**Comment by Zaxounette** [comment]
I have some wording nit picks, but before that I have a question:

It is said that the settings can't be modified anymore `in BTCPay Server`. This assumes that there is another way even once this action is taken.

First, is this the correct reading,
Second, what's the other way ?

**Comment by NicolasDorier** [comment]
@Zaxounette directly by SSH

**Comment by Zaxounette** [comment]
> @Zaxounette directly by SSH

Thanks ! 

Will review and send my wording nitpicks.

File: BTCPayServer/Controllers/ServerController.cs [hunk 1]
```diff
@@ -900,55 +903,88 @@ private bool CanAccessAuthorizedKeyFile()
...
+        public IActionResult SSHServiceDisable()
+        {
+            return View("Confirm", new ConfirmModel()
             {
-                TempData[WellKnownTempData.ErrorMessage] = exception.Message;
-            }
-            return RedirectToAction(nameof(SSHService));
+                Action = "Disable",
+                Title = "Disable modification of SSH settings",
+                Description = "This action is irreversible and will disable modification of SSH settings via BTCPay Server.",
```
Comment (Zaxounette) [inline BTCPayServer/Controllers/ServerController.cs:975]:
```suggestion
                Description = "This action is permanent and will remove the ability to change the SSH settings via the BTCPay Server user interface.",
```

File: BTCPayServer/Controllers/ServerController.cs [hunk 2]
```diff
@@ -900,55 +903,88 @@ private bool CanAccessAuthorizedKeyFile()
...
+            });
+        }
+        [Route("server/services/ssh/disable")]
+        [HttpPost]
+        public async Task<IActionResult> SSHServiceDisablePost()
+        {
+            var policies = await _SettingsRepository.GetSettingAsync<PoliciesSettings>() ?? new PoliciesSettings();
+            policies.DisableSSHService = true;
+            await _SettingsRepository.UpdateSetting(policies);
+            TempData[WellKnownTempData.SuccessMessage] = "SSH settings can't be modified anymore via BTCPay Server";
```
Comment (Zaxounette) [inline BTCPayServer/Controllers/ServerController.cs:986]:
```suggestion
            TempData[WellKnownTempData.SuccessMessage] = "Changes to the SSH settings are now permanently disabled in the BTCPay Server user interface";
```

File: BTCPayServer/Views/Server/SSHService.cshtml [hunk 3]
```diff
@@ -58,8 +57,24 @@
...
         </div>
     </div>
 }
+
+<h4 class="mt-5 mb-3">Other actions</h4>
+<div class="row">
+    <div class="col-md-8">
+        <div class="form-group">
+            <p>
+                <span>Increase the security of your instance by disabling the ability to change the SSH Settings via this BTCPay Server instance.<br /></span>
```
Comment (Zaxounette) [inline BTCPayServer/Views/Server/SSHService.cshtml:71]:
```suggestion
                <span>Increase the security of your instance by disabling the ability to change the SSH Settings in this BTCPay Server instance's user interface.<br /></span>
```

**Review by Zaxounette — COMMENTED** [review]
Some simple wording nit picks.

Mainly changing sentence structures and making it clear that it's only disabling the ability to change settings in the UI.
Also replacing the word `irreversible` by `permanent`.

**Review by Eskyee — APPROVED** [review]
Looks good to me

**Pull request merged**